AsyncDisplayKit/ASDisplayNode.h

```diff
@@ -506,7 +506,18 @@ typedef CALayer *(^ASDisplayNodeLayerBlock)();
 @interface ASDisplayNode (UIViewBridge)
 
 - (void)setNeedsDisplay;    // Marks the view as needing display. Convenience for use whether view is created or not, or from a background thread.
-- (void)setNeedsLayout;     // Marks the view as needing layout.  Convenience for use whether view is created or not, or from a background thread.
+
+/**
+ * Marks the view as needing layout. Convenience for use whether view is created or not, or from a background thread.
+ * 
+ * If this node was measured, calling this method triggers an internal relayout: the calculated layout is invalidated,
+ * and the supernode is notified or (if this node is the root one) a full measurement pass is executed using the old constrained size.
+ * 
+ * Note: If the relayout causes a change in size of the root node that is attached to a container view
+ * (table or collection view, for example), the container view must be notified to redraw. 
```

Comment:
"redraw" here should probably be "layout".

I am interested in the idea of formalizing the "container view" concept.  We should probably create a delegate API that the root node may implement to manage its interaction with the container.  This could then be used by ASTableView <-> ASCellNode, a new class like ASViewController <-> ASDisplayNode, etc.  If you haven't already done something like this, that's fine because it belongs as a separate effort (and I can probably take on that work).

Maybe revise the next line to say "For ASTableView and ASCollectionView, an empty batch editing transaction will be triggered to animate to new row / item sizes"  (because ASCollectionView doesn't actually have begin/end updates)